anvil/src/state.rs

```diff
@@ -287,6 +296,9 @@ impl<BackendData: Backend + 'static> AnvilState<BackendData> {
             *cursor_status3.lock().unwrap() = new_status;
         });
 
+        seat.text_input();
```

Comment:
Why is this call here? I looks a little out of place.

Reply:
It adds text input to the seat, should I rename it to add_text_input()?
I previously also used it to access the a handle, but I removed the need for that

Reply:
Or maybe `init_text_input`. Also please make sure, that other functions that need the userdata added by the function don't unwrap and crash, if the user has not called `init_text_input`. Also maybe add hints to their docs, that this call is required.

Reply:
I have this in the docs:

_This module provides you with utilities to handle input methods, it must be used in conjunction with the text input module to work._

`
seat.init_text_input();
// Add text input capabilities to a seat, needed for the input method to work
seat.add_input_method(XkbConfig::default(), 200, 25);
// Add input method capabilities to a seat
`

_This module provides you with utilities to handle text input surfaces, it must be used in conjunction with the input method module to work._

`
seat.init_text_input();
// Add text input capabilities to a seat
seat.add_input_method(XkbConfig::default(), 200, 25);
// Add input method capabilities to a seat, , needed for text input to work
`

Is that enough? :)